src/python/pants/bin/pants_runner.py

```diff
@@ -66,7 +72,7 @@ def run(self):
       warnings.filterwarnings(action='ignore', message=message_regexp)
 
     # TODO https://github.com/pantsbuild/pants/issues/7205
-    if global_bootstrap_options.enable_pantsd and not self.will_terminate_pantsd():
+    if self._should_run_with_pantsd(global_bootstrap_options):
       try:
         return RemotePantsRunner(self._exiter, self._args, self._env, options_bootstrapper).run()
       except RemotePantsRunner.Fallback as e:
```

Comment:
inlining of the LocalPantsRunner import can be moved to right before its used below